### PR TITLE
Fix iTranslate api error

### PR DESCRIPTION
**Pull request by danielzhang130**
Short language code like pt, zh is sent when iTranslate requires long ones like pt-PT and zh-CN

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

### CODE REVIEW

File: src/Translation/ItranslateApi.php [hunk 1]
```diff
@@ -81,14 +85,37 @@ private function transformLanguageCode(?string $language): ?string
...
+    foreach ($array_of_arrays as $array) {
+      foreach ($array as $element) {
+        $result[] = $element;
+      }
+    }
+
+    return $result;
+  }
+
+  private function chooseWithoutPoliticalBias(array $long_language_codes): string
```
Comment (danielzhang130) [inline src/Translation/ItranslateApi.php:118]: a bit of political humor here. If you think it has no place here, say so.

**Comment by danielzhang130** [comment]
Actually this would be blocking the google translate PR, so this will have to go in first

File: src/Translation/ItranslateApi.php (hunk 1 above)
Comment (JoshuaSchaefer) [reply src/Translation/ItranslateApi.php:118]: I think this is fine if you are all good with it 👍

**Review by JoshuaSchaefer — APPROVED** [review]
I am all good with these changes. Just make sure that this PR goes through before the Google API PR is posted as you mentioned already. Also, there might be a couple of updates to the Google API PR as well and maybe look into if the same problem occurs with the google API. Otherwise, looks good 👍

**Comment by danielzhang130** [comment]
i actually think this is not the solution. While it might be good to have resilience in this `transformLanguageCode` method for extending language codes, i think the solution to this particular problem is to move away from cookies. I'll make another pr

**Pull request closed without merging**